### ISSUE TITLE
Certificate warning does not work

### ISSUE THREAD

**Issue by luizluca** [browsh-org/browsh#127]
When a SSL certificate is not trusted, browsh does not allow me to accept and continue, as firefox does. I need to open gui and do it using X

**Comment by tombh** [comment]
Yeah this is big drawback of my current approach with Browsh. The technical reason is that Webextensions don't allow injection of JS into non-web pages.

This is related to #106

**Comment by tobimensch** [comment]
@tombh 
Maybe compiling a version of firefox without certificate warnings and the like would be a solution. One layer of security would be missing though.

**Comment by luizluca** [comment]
If you disable certificate warning, would it be possible to known afterwards if the page certificate was invalid? You could add a warning 'popup'.

Another approach:  If you disable certificate warning, would it be possible to intercept the URL requests before firefox does the SSL checking? browsh could check it by itself (just like firefox would do) and, if needed, show the warnings. If user accepts or cert is good, let firefox access the URL.

**Comment by tombh** [comment]
@tobimensch Excellent idea! I never considered that. We shouldn't need to compile, it should be possible through preferences. What's more such an approach could work with some of the other problems we have, like the download manager dialog in #126.

@luizluca For sure! We can use `curl` or something in the background to do the certificate check for us, then show the result to the user.

**Comment by jypma** [comment]
Isn't that too late? If you disable the warning, firefox will already have made a connection (potentially from an AJAX trying to save sensitive data). Showing the warning to the user afterwards may have already leaked data...

**Comment by tobimensch** [comment]
@jypma 
The interfacer written in Go could check for any certificates before the webextension and therefore firefox is used.

**Comment by empjustine** [comment]
Instead of opening a major security liability in all HTTPS use, I think a way to override the user profile settings for trusted invalid certificates seems more reasonable.

From the docs:

Security certificate settings: The cert9.db file stores all your security certificate settings and any SSL certificates you have imported into Firefox.

**Comment by luizluca** [comment]
@empjustine , maybe adding the cert before the first connection and removing it just after the first connection would be enough for the session. Does Firefox check cert validity on every request?

**Comment by frakman1** [comment]
So does this mean I can't use `browsh` to access a LAN device with a self-signed certificate? 
My error looks like this:

``` 
WEBEXT: Tab undefined received new focus
WEBEXT: New tab created: [object Object]
FF-MRNT: 303:[1,0,{"error":"session not created","message":"Error: NS_ERROR_GENERATE_FAILURE(NS_ERROR_MODULE_SECURITY, MOZILLA_PKIX_ERROR_SELF_SIGNED_CERT)","stacktrace":"#checkLoadingState@chrome://remote/content/shared/Navigate.jsm:209:28\nonStateChange@chrome://remote/content/shared/Navigate.jsm:254:28\n"},null]
```
